swap_meet/item.py

```
@@ -1,2 +1,31 @@
+import uuid
+
+CONDITION_DESCRIPTIONS = {
+    0: "Acceptable",
+    1: "Good",
+    2: "Very Good",
+    3: "Excellent",
+    4: "Like New",
+    5: "New"
+}
+
+
 class Item:
-    pass
\ No newline at end of file
+    def __init__(self, id=None, condition=0, age=0):
+        self.id = uuid.uuid4().int if id is None else id
+        self.condition = condition
+        self.age = age  # age in years
+
+    def get_category(self):
+        return self.__class__.__name__
+
+    def __str__(self):
+        return f"An object of type {self.get_category()} with id {self.id}."
+
+    def condition_description(self):
+        condition = self.condition // 1
+
+        if condition not in CONDITION_DESCRIPTIONS:
+            return "Condition Out of Range"
```

There wasn't a clear specification for what to do for the description if the condition was outside the expected range. As a result returning a canned string is fine.

However, from the way the string is phrased, it sounds more like we're treating this an error case versus a more neutral response like `"condition unknown"`. If we are thinking of this case as an error, then we should prefer to raise an error instead. We might also add validation to the constructor to ensure the condition is within the valid range so that a caller would have the information that they supplied an invalid condition as early as possible, rather than potentially having invalid data affect other parts of the system.